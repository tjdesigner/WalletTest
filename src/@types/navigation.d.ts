/**
 * Learn more about using TypeScript with React Navigation:
 * https://reactnavigation.org/docs/typescript/
 */

import { ParamListBase } from "@react-navigation/native"
import { NativeStackScreenProps } from "@react-navigation/native-stack"

import type {
  CompositeScreenProps,
  NavigatorScreenParams,
} from "@react-navigation/native"
import type { StackScreenProps } from "@react-navigation/stack"

interface Card {
  id?: string
  cardNumber: string
  name: string
  expirationDate: string
  cvv: string
}

/****************************************************/

export type RootStackParamList = {
  Home: Card | undefined

  Register:
    | {
        card?: Card
      }
    | undefined

  RegisterConfirmation:
    | {
        card: Card
      }
    | undefined

  LoadingAnimation: undefined

  MyCards:
    | {
        allCards?: Card[]
      }
    | undefined
}

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type RootStackScreenProps<ScreenName extends keyof RootStackParamList> =
  NativeStackScreenProps<RootStackParamList, ScreenName>
