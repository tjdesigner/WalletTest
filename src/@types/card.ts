export interface Card {
  id?: string
  cardName?: string
  cardNumber?: string
  name?: string
  expirationDate?: string
  cvv?: string
}
