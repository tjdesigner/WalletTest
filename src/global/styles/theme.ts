export default {
  colors: {
    black: "black",
    white: "white",
    danger: "#DC1637",
    info: "#12C2E9",
    warning: "#FFE900",
    success: "#2ec64e",
    background: "rgba(238, 238, 238, .2)",
    primary: "#142995",
    secondary: "#12C2E9",
    tertiary: "#A5FF32",
    backgroundDisabled: "#EEEEEE",
    textDisabled: "#BBBBBB",
  },

  fonts: {
    family: {
      regular: "PTSans-Regular",
      italic: "PTSans-Italic",
      boldItalic: "PTSans-BoldItalic",
      bold: "PTSans-Bold",
    },
    weight: {
      normal: "normal",
      bold: "bold",
      l1: "100",
      l2: "200",
      l3: "300",
      l4: "400",
      l5: "500",
      l6: "600",
      l7: "700",
      l8: "800",
      l9: "900",
      undefined,
    },
  },
  fontSize: {
    xxxs: `${1}px`,
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    default: `${16}px`,
    medium: `${24}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${60}px`,
    veryLarge: `${76}px`,
    titlePage: `${20}px`,
    title: `${28}px`,
    specific: `${18}px`,
  },
  fontSizeNumber: {
    xxxs: 1,
    xxs: 2,
    xs: 4,
    small: 8,
    default: 16,
    medium: 24,
    large: 32,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
    titlePage: 20,
    title: 28,
    specific: 18,
  },
  radius: {
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    small2: `${12}px`,
    default: `${16}px`,
    medium: `${20}px`,
    medium2: `${24}px`,
    medium3: `${28}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${54}px`,
    veryLarge: `${76}px`,
  },
  radiusNumber: {
    xxs: 2,
    xs: 4,
    small: 8,
    small2: 12,
    default: 16,
    medium: 24,
    large: 32,
    large2: 36,
    large3: 40,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
  },
  spaces: {
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    default: `${16}px`,
    medium: `${20}px`,
    medium2: `${24}px`,
    medium3: `${28}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${54}px`,
    veryLarge: `${76}px`,
  },
  spacesNumber: {
    xxs: 2,
    xs: 4,
    small: 8,
    default: 16,
    medium: 24,
    large: 32,
    large2: 36,
    large3: 40,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
  },
}
