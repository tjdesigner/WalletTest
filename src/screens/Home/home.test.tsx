import React from "react";
import { Platform } from 'react-native'
import { fireEvent, render } from '@testing-library/react-native'
import { HomeScreen } from ".";
import { RootStackParamList } from "../../@types/navigation";
import type { NativeStackScreenProps } from "@react-navigation/native-stack";
import { ThemeProvider } from "styled-components";
import theme from "../../global/styles/theme";

const navigateMock = jest.fn()

type HomeScreenProps = NativeStackScreenProps<RootStackParamList, 'Home'>

const optionsNavigation = {
    navigation: {
        navigate: navigateMock
    }
} as unknown as HomeScreenProps

describe('Home', () => {
    it('should render home screen', () => {
        const { toJSON } = render(<ThemeProvider theme={theme}>
            <HomeScreen {...optionsNavigation} />
        </ThemeProvider>)
        expect(toJSON()).toMatchSnapshot()

    })

    it('should click button my cards', async () => {
        Platform.OS = 'android'

        const { findByTestId } = render(<ThemeProvider theme={theme}>
            <HomeScreen {...optionsNavigation} />
        </ThemeProvider>)

        const myCardButton = await findByTestId('mycard-testid')
        fireEvent.press(myCardButton)

        expect(navigateMock).toHaveBeenCalled()
    })

    it('should click button register', async () => {
        Platform.OS = 'ios'
        const { findByTestId } = render(<ThemeProvider theme={theme}>
            <HomeScreen {...optionsNavigation} />
        </ThemeProvider>)

        const myCardButton = await findByTestId('register-testid')
        fireEvent.press(myCardButton)

        expect(navigateMock).toHaveBeenCalled()
    })
})