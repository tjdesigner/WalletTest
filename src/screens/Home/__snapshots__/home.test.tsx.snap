// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render home screen 1`] = `
<View
  style={
    {
      "backgroundColor": "#142995",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingTop": 17,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "rgba(238, 238, 238, .2)",
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "height": 370,
        "left": -115,
        "position": "absolute",
        "top": -205,
        "transform": [
          {
            "rotate": "58deg",
          },
        ],
        "width": 370,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "rgba(238, 238, 238, .2)",
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "bottom": -205,
        "height": 370,
        "position": "absolute",
        "right": -115,
        "transform": [
          {
            "rotate": "58deg",
          },
        ],
        "width": 370,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 64,
      }
    }
  >
    <Text
      color="white"
      marginBottom={32}
      style={
        {
          "color": "white",
          "fontFamily": "PTSans-Regular",
          "fontSize": 28,
          "marginBottom": 32,
        }
      }
      text="Wallet Test"
      type="contentPage"
    >
      Wallet Test
    </Text>
    <View
      marginBottom={16}
      marginLeft={24}
      marginRight={16}
      style={
        {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "backgroundColor": "#142995",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginBottom": 16,
          "marginLeft": 24,
          "marginRight": 16,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#12C2E9",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "height": 54,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="mycard-testid"
      >
        <Text
          backgroundColor="secondary"
          color="white"
          fullWidth={true}
          onPress={[Function]}
          style={
            {
              "backgroundColor": "#12C2E9",
              "color": "white",
              "fontFamily": "PTSans-Regular",
              "fontSize": 18,
              "fontWeight": "100",
            }
          }
          suppressHighlighting={true}
          testIDButton="mycard-testid"
          textButton="meus cart천es"
        >
          meus cart천es
        </Text>
      </View>
    </View>
    <View
      marginBottom={16}
      marginLeft={24}
      marginRight={16}
      style={
        {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "backgroundColor": "#142995",
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginBottom": 16,
          "marginLeft": 24,
          "marginRight": 16,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#A5FF32",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "height": 54,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="register-testid"
      >
        <Text
          backgroundColor="tertiary"
          fullWidth={true}
          onPress={[Function]}
          style={
            {
              "backgroundColor": "#A5FF32",
              "color": "#142995",
              "fontFamily": "PTSans-Regular",
              "fontSize": 18,
              "fontWeight": "100",
            }
          }
          suppressHighlighting={true}
          testIDButton="register-testid"
          textButton="cadastrar cart찾o"
        >
          cadastrar cart찾o
        </Text>
      </View>
    </View>
  </View>
</View>
`;
